### DISCUSSION TITLE
Access to FPC/Plugin-Data in an FLP

### DISCUSSION THREAD

**Discussion by SilverShepherd** [demberto/PyFLP#192] [Q&A]
Hello,
I am a bit new to pyflp and not to deep into programming.
(I learned programming for myself to use the FL Midi-Scripting Feature and that is the most experience I have)

PyFLP is a bit new and difficult for me to use. 
I already found out how to change the Project tempo, how to rename Channels and how to set all the parameters arround a Channel-Plugin.
My Goal is now to understand the **data of the FPC-Plugin in a FLP-File** to create an converter from current MPC-Sample-Kits (from Akai) to FPC.

I already read the documentation several times. And now I finally found the over 28000 bytes that seem to be the raw FPC Data via FLPEdit
(I found out that Plugin Data is always in the List of "ID_Plugin_Parameters Byte Array]" and not in the List of "ID_Plugin_New Byte Array") ;)

Now the question: How can I get access to the FPC-Data (in the FLP file) I found by analyzing the FLP with FLPEdit, 
via PyFLP and how can I manipulate and save it?
A little example code would be helpful.

And what I'm also wondering about:
Where are all the possible commands listet for PyFLP?
In the example are commands used which aren't written down anywhere.
I kind of understand the varibales, that are mentioned in the documentation but the commands weren't findable for me.

**Comment by demberto** [comment]
The `PluginID.Data` event stores that raw data, if not parsed by a plugin parser already. The events attribute contains all events associated to a model (like Channel, Mkxer etc)
